astropy/visualization/log_linear_rgb.py

```diff
@@ -0,0 +1,380 @@
+        Intensity that should be mapped to white (a scalar or
+        array of R, G, B). If `None`, each image's maximum value is used.
+    scalea : float, optional
+        Log scaling exponent.
+    filename : str, optional
+        Write the resulting RGB image to a file (file type determined
+        from extension).
+    output_image_format : numpy scalar type, optional
+        Image output format. Default is np.uint8.
+
+    Returns
+    -------
+    rgb : ndarray
+        RGB (either float or integer with 8-bits per channel) color image
+        as an NxMx3 numpy array.
+
+    Notes
+    -----
+    This procedure of clipping and then scaling is similar to the DS9
+    image algorithm (see the DS9 reference guide:
+    http://ds9.si.edu/doc/ref/how.html).
+
+    """
+    return make_rgb(
+        image_r,
+        image_g,
+        image_b,
+        minimum=minimum,
+        maximum=maximum,
+        stretch=LogStretch(a=scalea),
+        filename=filename,
+        output_image_format=output_image_format,
+    )
+
+
+def make_linear_rgb(
+    image_r,
+    image_g,
+    image_b,
+    minimum=None,
+    maximum=None,
+    filename=None,
+    output_image_format=np.uint8,
+):
+    """
+    Return a Red/Green/Blue color image from 3 images using a linear stretch.
+    Includes optional clipping of the input values before scaling.
+
+    The input images can be int or float, and in any range or bit-depth,
+    but must have the same shape (NxM).
+
+    For a more detailed look at the use of this method, see the document
+    :ref:`astropy:astropy-visualization-rgb`.
+
+    Parameters
+    ----------
+    image_r : ndarray
+        Image to map to red.
+    image_g : ndarray
+        Image to map to green.
+    image_b : ndarray
+        Image to map to blue.
+    minimum : float or array-like, optional
+        Intensity that should be mapped to black (a scalar or
+        array of R, G, B). If `None`, each image's minimum value is used.
+    maximum : float or array-like, optional
+        Intensity that should be mapped to white (a scalar or
+        array of R, G, B). If `None`, each image's maximum value is used.
+    filename : str, optional
+        Write the resulting RGB image to a file (file type determined
+        from extension).
+    output_image_format : numpy scalar type, optional
+        Image output format. Default is np.uint8.
+
+    Returns
+    -------
+    rgb : ndarray
+        RGB (either float or integer with 8-bits per channel) color image
+        as an NxMx3 numpy array.
+
+    Notes
+    -----
+    This procedure of clipping and then scaling is similar to the DS9
+    image algorithm (see the DS9 reference guide:
+    http://ds9.si.edu/doc/ref/how.html).
+
+    """
+    return make_rgb(
+        image_r,
+        image_g,
+        image_b,
+        minimum=minimum,
+        maximum=maximum,
+        stretch=LinearStretch(),
+        filename=filename,
+        output_image_format=output_image_format,
+    )
```

Comment:
Seems a bit too much to have a class and 3 functions to do basically the same thing. But I will leave that choice to visualization maintainers ;).

Reply:
A very fair point. My thought was to have convenience functions parallel to the "familiar" ``make_lupton_rgb()`` function, falling back on the underlying class. However, I also see the merit in moving away from proliferating convenience functions and towards more explicit user-required input of a unified convenience function (eg, ``make_rgb()``).  Potentially ``make_lupton_rgb()`` could also be reworked to require a BaseStretch subclass input, and depreciation warnings can be raised for constant stretch value or Q keyword arguments. This would limit the RGB visualization to two classes (one for "scale R,G,B separately" and one for the Lupton "scale with interconnected RGB") and two helper functions. Looking forward to thoughts from the maintainers!